### DISCUSSION TITLE
Using Fluid with React

### DISCUSSION THREAD

**Discussion by tylerbutler** [microsoft/FluidFramework#5462] [Documentation topics]
~This article isn't written yet, but you can discuss and vote on it here. We prioritize writing articles that receive the most votes.~

This article is now published here: <https://fluidframework.com/docs/recipes/react/>

**Comment by Vivihung** [comment]
The example [here](https://github.com/microsoft/FluidFramework/issues/5478) might be helpful. 
Couple suggestions:
**React component flavors**
1. How to use Fluid with React functional components (for the React hooks group)
2. How to use Fluid with React class components (Fluid has couple React component interfaces that can be demonstrated here)

**Scenario focus**
1. How to handle the event raised by Fluid object and connect it to React component state to trigger re-rendering
2. Test examples: 
    * How to mock Fluid object for testability.
    * How to mock Fluid service dependency to support functional testing (e.g., React snapshots)

Notes:
Collaboration isn't trivial to test in general. We should look around the existing tools and find a good recipe for Fluid dev.
